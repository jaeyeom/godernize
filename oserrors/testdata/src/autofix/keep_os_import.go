package autofix

import (
	"fmt"
	"os"
)

var _ = keepOSImport

func keepOSImport() {
	file, err := os.Open("test.txt")
	if err != nil {
		if os.IsNotExist(err) { // want `os.IsNotExist is deprecated, use errors.Is\(err, fs.ErrNotExist\) instead`
			fmt.Println("File does not exist")
		}
	}
	defer file.Close()

	// os package is still used for other purposes
	info, _ := os.Stat("test.txt")
	fmt.Println(info.Size())
}
