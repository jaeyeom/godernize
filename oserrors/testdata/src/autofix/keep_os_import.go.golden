package autofix

import (
	"fmt"
	"os"
)

var _ = keepOSImport

func keepOSImport() {
	file, err := os.Open("test.txt")
	if err != nil {
		if errors.Is(err, fs.ErrNotExist) { // want `os.IsNotExist is deprecated, use errors.Is\(err, fs.ErrNotExist\) instead`
			fmt.Println("File does not exist")
		}
	}
	defer file.Close()

	// os package is still used for other purposes
	info, _ := os.Stat("test.txt")
	fmt.Println(info.Size())
}